src/streams/appendToStream/index.ts

```diff
@@ -0,0 +1,77 @@
+import { StreamsService } from "../../../generated/streams_grpc_pb";
+
+import { Client } from "../../Client";
+import { ANY } from "../../constants";
+import {
+  BaseOptions,
+  AppendResult,
+  AppendExpectedRevision,
+  EventData,
+} from "../../types";
+
+import { append } from "./append";
+import { batchAppend } from "./batchAppend";
+
+export interface AppendToStreamOptions extends BaseOptions {
+  /**
+   * Asks the server to check the stream is at specific revision before writing events.
+   * @default ANY
+   */
+  expectedRevision?: AppendExpectedRevision;
+  /**
+   * The batch size, in bytes.
+   * @default 3 * 1024 * 1024
+   */
+  batchAppendSize?: number;
+}
+
+export type InternalAppendToStreamOptions = Required<
+  Omit<AppendToStreamOptions, keyof BaseOptions>
+> &
+  BaseOptions;
+
+declare module "../../Client" {
+  interface Client {
+    /**
+     * Appends events to a given stream.
+     * @param streamName A stream name.
+     * @param events Events or event to write.
+     * @param options Writing options.
+     */
+    appendToStream(
+      streamName: string,
+      events: EventData | EventData[],
+      options?: AppendToStreamOptions
+    ): Promise<AppendResult>;
+  }
+}
+
+Client.prototype.appendToStream = async function (
+  this: Client,
+  streamName: string,
+  event: EventData | EventData[],
+  {
+    expectedRevision = ANY,
+    batchAppendSize = 3 * 1024 * 1024,
+    ...baseOptions
+  }: AppendToStreamOptions = {}
+): Promise<AppendResult> {
+  const events = Array.isArray(event) ? event : [event];
+
+  if (
```

Comment:
See comment above: https://github.com/EventStore/EventStore-Client-NodeJS/pull/229/files#r741774656

Reply:
For context:

In batch append, the stream was opened with default credentials, and we cannot add credentials to the specific append batch. Therefore, we fall back to a single append if per call credentials are passed.